### PR TITLE
Automatically enable memory only keys when payload size of reconnectapp/openapp is too large

### PR DESCRIPTION
**Pull request by tgolen**
cc @marcaaron 

### Details
This PR will listen for the server attempting to update the memory only keys feature. If there is an onyx update for that key, then the frontend will fully enable the feature.

### Fixed Issues
Part of https://github.com/Expensify/Expensify/issues/300442

### Tests
**Note: this can only be fully tested by an internal engineering using the PR https://github.com/Expensify/Web-Expensify/pull/38491**

#### Tests for anyone (web and desktop only)
1. Sign onto NewDot
2. Run `enableMemoryOnlyKeys()` from the JS console
3. Verify the logs say `[MemoryOnlyKeys] enabled`
4. Run `disableMemoryOnlyKeys()` from the JS console
5. Verify the logs say `[MemoryOnlyKeys] disabled`

#### Tests for an internal engineer (all platforms except mobile web)
1. Update Web-Expensify to use the branch from https://github.com/Expensify/Web-Expensify/pull/38491
2. Set the `MEMORYONLYKEYS_THRESHOLD` constant to be a negative number (like `-1`)
3. Sign out from NewDot
4. Sign in to NewDot
5. Verify the logs show `[MemoryOnlyKeys] enabled`
6. Go to JS console to look in Onyx
7. Verify that no `report_` keys and no `policy_` keys are saved

- [ ] Verify that no errors appear in the JS console

### Offline tests
None

### QA Steps
You can perform the tests above in the `Tests for anyone (web and desktop only)` section

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome **Note: this cannot be tested on mobile web due to not being able to access the JS console**
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari  **Note: this cannot be tested on mobile web due to not being able to access the JS console**
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

image

</details>

<details>
<summary>iOS</summary>

image

</details>

<details>
<summary>Android</summary>

image

</details>

### CODE REVIEW

File: src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../../ONYXKEYS';
+
+const memoryOnlyKeys = [ONYXKEYS.COLLECTION.REPORT, ONYXKEYS.COLLECTION.POLICY, ONYXKEYS.PERSONAL_DETAILS_LIST];
+
+const enable = () => {
+    console.log('Turning on memory only keys');
```
Comment (venture1981) [inline src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js:7]: accrued lint error.
As you can see, there is `Unexpected console statement` error

File: src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    Onyx.set(ONYXKEYS.IS_USING_MEMORY_ONLY_KEYS, true);
+    Onyx.setMemoryOnlyKeys(memoryOnlyKeys);
+};
+
+const disable = () => {
+    console.log('Turning off memory only keys');
```
Comment (venture1981) [inline src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js:13]: same here.

**Review by venture1981 — COMMENTED** [review]
put comments about github lint errors.

**Comment by marcaaron** [comment]
@tgolen Maybe you are on top of this already (sorry if so), but I think we're gonna want to add a test that if `OpenApp` has the memory only keys directive in the response then none of the reports, policies, etc keys should be saved to storage. 

The original intention of the mode was that it would run before `OpenApp` (and sign in) so after the response returns it won't spend time saving things to storage. Can look at the IndexedDB in the ChromeDevTools to figure that out for the purposes of this PR. We shouldn't see any reports or policies in storage.

**Comment by tgolen** [comment]
Thanks for spelling that out. I definitely didn't have that context, but it appears like this is already being handled: https://github.com/Expensify/App/blob/0a21797bbfcd48e8235dc2b7ff160f0c2d7cb6fa/src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js#L4

**Comment by tgolen** [comment]
It sounds like maybe I just need to update the testing instructions

**Comment by tgolen** [comment]
OK, I updated the tests and retested it. Thanks for noticing that!

**Comment by tgolen** [comment]
Bump for review @arosiclair

**Comment by arosiclair** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by arosiclair** [comment]
LGTM though I'm still blocked on Android build issues so I can't test there atm

File: src/libs/Middleware/SaveResponseInOnyx.js [hunk 3]
```diff
@@ -15,6 +18,17 @@ function SaveResponseInOnyx(response, request) {
             return;
         }
 
+        // Supports both the old format and the new format
```
Comment (marcaaron) [inline src/libs/Middleware/SaveResponseInOnyx.js:21]: We could possibly add some brief explanation of which one is the old vs new? 

NAB/just curious - How did things end up in this state? Is this temporary or are we going to support these two formats forever?

File: src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../../ONYXKEYS';
+
+const memoryOnlyKeys = [ONYXKEYS.COLLECTION.REPORT, ONYXKEYS.COLLECTION.POLICY, ONYXKEYS.PERSONAL_DETAILS_LIST];
+
+const enable = () => {
+    console.debug('[MemoryOnlyKeys] enabled');
```
Comment (marcaaron) [inline src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js:7]: Maybe this (and the one belwo) could be a `Log.info()`? Could help show that someone enabled this mode if we are investigating server logs.

File: src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import * as MemoryOnlyKeys from '../MemoryOnlyKeys';
+
+const exposeGlobalMethods = () => {
```
Comment (marcaaron) [inline src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.js:3]:
```suggestion
const exposeGlobalMemoryOnlyKeysMethods = () => {
```

File: src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.native.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+/**
+ * This is a no-op because the global methods will only work for web and desktop
+ */
+const exposeGlobalMethods = () => {};
```
Comment (marcaaron) [inline src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.native.js:4]:
```suggestion
const exposeGlobalMemoryOnlyKeysMethods = () => {};
```

File: src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.native.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+
+export default exposeGlobalMethods;
```
Comment (marcaaron) [inline src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.native.js:6]:
```suggestion
export default exposeGlobalMemoryOnlyKeysMethods;
```

File: src/libs/actions/User.js [hunk 8]
```diff
@@ -568,6 +569,16 @@ function subscribeToUserEvents() {
...
             }
+
+            // If there is an OnyxUpdate for using memory only keys, enable them
+            _.each(pushJSON, ({key, value}) => {
+                if (key !== ONYXKEYS.IS_USING_MEMORY_ONLY_KEYS || !value) {
+                    return;
+                }
+
+                MemoryOnlyKeys.enable();
+            });
```
Comment (marcaaron) [inline src/libs/actions/User.js:580]: NAB, I thought maybe we were just sending these `onyxUpdate` with this key via https? Is including the check here for Pusher updates sort of a future proofing?

**Review by marcaaron — CHANGES_REQUESTED** [review]
Couple small things, questions, overall - looks great

File: src/libs/Middleware/SaveResponseInOnyx.js [hunk 9]
```diff
@@ -15,6 +18,17 @@ function SaveResponseInOnyx(response, request) {
...
+        const onyxUpdates = _.isArray(responseData) ? responseData : responseData.onyxData;
+        // If there is an OnyxUpdate for using memory only keys, enable them
+        _.each(onyxUpdates, ({key, value}) => {
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/libs/Middleware/SaveResponseInOnyx.js:24]: Great idea

File: src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js (hunk 4 above)
Comment (tgolen) [reply src/libs/actions/MemoryOnlyKeys/MemoryOnlyKeys.js:7]: That's a good idea

File: src/libs/actions/User.js (hunk 8 above)
Comment (tgolen) [reply src/libs/actions/User.js:580]: Yeah, you're right. This is a hold-over from my original testing/implementation but it doesn't need to be here and is a little confusing. I'll remove it.

**Comment by tgolen** [comment]
Updated!

File: src/libs/Middleware/SaveResponseInOnyx.js (hunk 3 above)
Comment (marcaaron) [reply src/libs/Middleware/SaveResponseInOnyx.js:30]: Ah makes sense. I need to brush up on reliable updates.

Comment (marcaaron) [reply src/libs/Middleware/SaveResponseInOnyx.js:30]: Thanks for the detailed explanation!

**Review by arosiclair — DISMISSED** [review]
Android tests well 👍

File: src/libs/actions/User.js [hunk 10]
```diff
@@ -19,6 +19,7 @@ import * as ErrorUtils from '../ErrorUtils';
 import * as Session from './Session';
 import * as PersonalDetails from './PersonalDetails';
 import * as OnyxUpdates from './OnyxUpdates';
+import * as MemoryOnlyKeys from './MemoryOnlyKeys/MemoryOnlyKeys';
```
Comment (marcaaron) [inline src/libs/actions/User.js:22]:
```suggestion
```

**Review by marcaaron — DISMISSED** [review]
LGTM (but there's a lint issue to address)

File: src/libs/actions/User.js (hunk 10 above)
Comment (tgolen) [reply src/libs/actions/User.js:22]: Ah, thanks

**Comment by tgolen** [comment]
Updated! Sorry, all your reviews got dismissed. Next one to approve, please feel free to merge.

**Review by arosiclair — APPROVED** [review]

File: src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import * as MemoryOnlyKeys from '../MemoryOnlyKeys';
+
+const exposeGlobalMemoryOnlyKeysMethods = () => {
+    window.enableMemoryOnlyKeys = () => {
+        MemoryOnlyKeys.enable();
+    };
+    window.disableMemoryOnlyKeys = () => {
+        MemoryOnlyKeys.disable();
+    };
```
Comment (cubuspl42) [inline src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.js:9]: What is it for? Hand-debugging via console, or what?

Comment (cubuspl42) [reply src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.js:9]: We're just [adding typing](https://github.com/Expensify/App/pull/32164#discussion_r1424192764) for this part and I'm wondering if it's actually beneficial to extend the `Window` interface, or maybe we don't need any typing for this if we don't intend to use it programmatically.

Comment (marcaaron) [reply src/libs/actions/MemoryOnlyKeys/exposeGlobalMemoryOnlyKeysMethods/index.js:9]: Yes, it's a JS console thing. And no, it shouldn't really be used in the code anywhere. So I agree there is no point in getting crazy with the types for this.
